mirascope/core/base/_utils/_get_call_args_field_names_and_validate.py

```diff
@@ -0,0 +1,26 @@
+import inspect
+from collections.abc import Callable
+
+from pydantic import BaseModel
+
+from mirascope.core.base.from_call_args import is_from_call_args
+
+
+def get_call_args_field_names_and_validate(
+    response_model: object, fn: Callable
+) -> set[str]:
+    if not (inspect.isclass(response_model) and issubclass(response_model, BaseModel)):
+        return set()
+    call_args_fields = {
+        name
+        for name, field in response_model.model_fields.items()
+        if is_from_call_args(field)
+    }
+
+    signature = inspect.signature(fn)
+    arguments = signature.parameters.keys()
```

Comment:
this is not correct. please see my other comment about combining the methods and using `get_fn_args` here because that method will properly get `kwargs` as well

Reply:
to be clear, this is why we need to merge these methods. we won't have all the field names until runtime

Reply:
>  because that method will properly get kwargs as well

Do we need to target the kwargs defined in the function?
My understanding is that we need to check if the pydantic field names marked with `FromCallArgs()` are included in the call function argument names.

Finally, `kwargs` is checked for correct arguments as a standard Python function at call time.

```python
class BookRecommendation(BaseModel):
    genre: Annotated[str, FromCallArgs()]
    title: str = Field(..., description="The title of the recommended book")
    author: str = Field(..., description="The author of the recommended book")

@openai.call("gpt-4o-mini", response_model=BookRecommendation)
def recommend_book(genre: str) -> str:
    return f"Recommend a {genre} book"

kwargs = {"genre": "fantasy"}
ng_kwargs = {"invalid": "fantasy"}
print(recommend_book(**kwargs)) # OK
print(recommend_book(**ng_kwargs)) # NG
```

Output:
```bash
genre='fantasy' title='The Name of the Wind' author='Patrick Rothfuss'
Traceback (most recent call last):
  File "/Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_49.py", line 18, in <module>
    print(recommend_book(**ng_kwargs)) # NG
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_extract.py", line 135, in inner
    call_response = create_decorator(fn=fn, **create_decorator_kwargs)(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_create.py", line 173, in inner
    fn_args = get_fn_args(fn, args, kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_utils/_get_fn_args.py", line 13, in get_fn_args
    bound_args = signature.bind_partial(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/.local/share/uv/python/cpython-3.12.5-macos-aarch64-none/lib/python3.12/inspect.py", line 3280, in bind_partial
    return self._bind(args, kwargs, partial=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/.local/share/uv/python/cpython-3.12.5-macos-aarch64-none/lib/python3.12/inspect.py", line 3262, in _bind
    raise TypeError(
TypeError: got an unexpected keyword argument 'invalid'

```